### PR TITLE
gsclient-go 2.0 compatibility

### PR DESCRIPTION
**Pull request by nvthongswansea**

### CODE REVIEW

File: gridscale/resource_gridscale_server.go [hunk 1]
```diff
@@ -359,11 +356,36 @@ func resourceGridscaleServerCreate(d *schema.ResourceData, meta interface{}) err
 		Cores:           d.Get("cores").(int),
 		Memory:          d.Get("memory").(int),
 		LocationUUID:    d.Get("location_uuid").(string),
-		HardwareProfile: d.Get("hardware_profile").(string),
 		AvailablityZone: d.Get("availability_zone").(string),
 		Labels:          convSOStrings(d.Get("labels").(*schema.Set).List()),
 	}
 
+	profile := d.Get("hardware_profile").(string)
+	if profile == "legacy" {
```
Comment (itakouna) [inline gridscale/resource_gridscale_server.go:364]: I think using the switch statement could be better!!

File: gridscale/resource_gridscale_server.go [hunk 2]
```diff
@@ -536,7 +536,7 @@ func resourceGridscaleServerUpdate(d *schema.ResourceData, meta interface{}) err
 
 	//The ShutdownServer command will check if the server is running and shut it down if it is running, so no extra checks are needed here
 	if shutdownRequired {
-		err = client.ShutdownServer(emptyCtx, d.Id())
+		err = client.StopServer(emptyCtx, d.Id())
```
Comment (itakouna) [inline gridscale/resource_gridscale_server.go]: Why did you replace `ShutdownServer` with `StopServer`? `StopServer` sends an ACPI shutdown signal to the virtual machine, which does like the power button of a computer. `ShutdownServer` will do graceful power-off.

**Comment by itakouna** [comment]
We need a test case for the issues https://github.com/terraform-providers/terraform-provider-gridscale/issues/12 and https://github.com/terraform-providers/terraform-provider-gridscale/issues/13.

File: gridscale/resource_gridscale_server.go (hunk 2 above)
Comment (nvthongswansea) [reply gridscale/resource_gridscale_server.go]: Currently ACPI shutdown does not work (I've tested)

Comment (itakouna) [reply gridscale/resource_gridscale_server.go]: plz, keep `ShutdownServer`.  I tested it from the panel and worked well. if you have any error, report that in the developers' slack channel.

**Comment by nvthongswansea** [comment]
> #13

Tests are successful.

**Comment by itakouna** [comment]
> > #13
> 
> Tests are successful.

nice, but I meant to add (decrease resource) test case in https://github.com/terraform-providers/terraform-provider-gridscale/blob/master/gridscale/resource_gridscale_server_test.go

File: gridscale/resource_gridscale_server.go (hunk 2 above)
Comment (nvthongswansea) [reply gridscale/resource_gridscale_server.go]: Decreasing number of CPUs just needs the server to be stopped btw. I can't find any reasons why we need to shut it down when we just need to stop it.

Comment (itakouna) [reply gridscale/resource_gridscale_server.go]: both will stop the server. `StopServer ` does not consider the running processes. If a vm is writing to disk when it receives a Power Off command, data corruption may occur.

**Review by itakouna — APPROVED** [review]

**Pull request merged**